embassy-rp/src/clocks.rs

```diff
@@ -196,3 +196,40 @@ unsafe fn configure_pll(p: pac::pll::Pll, refdiv: u32, vco_freq: u32, post_div1:
     // Turn on post divider
     p.pwr().modify(|w| w.set_postdivpd(false));
 }
+
+/// Random number generator based on the ROSC RANDOMBIT register.
+///
+/// This will not produce random values if the ROSC is stopped or run at some
+/// harmonic of the bus frequency. With default clock settings these are not
+/// issues.
+pub struct RoscRng;
+
+impl RoscRng {
+    fn next_u8() -> u8 {
+        let random_reg = pac::ROSC.randombit();
+        let mut acc = unsafe { random_reg.read().randombit() as u8 };
```

Comment:
to avoid reading the reg in 2 places, you could start with `0` and loop for `0..u8::BITS`

Reply:
I was about to say that that would not work due to too many shifts, but I just realized that's only the case if the shift is after the access. Once again I am reminded that coding should not be done while sleepy.